### PR TITLE
Add clarification about reason phrases and HTTP/2

### PR DESCRIPTION
**Pull request by sbont**

### CODE REVIEW

File: content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ The `HttpResponse` entity has the following attributes:
 
 For more information on HTTP status codes, see the [W3C Specification of Status Code Definitions](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html).
 
-Setting `ReasonPhrase` has no effect; the runtime always uses the default reason phrase for the status code.
+Setting `ReasonPhrase` has no effect; for HTTP/1.x requests the runtime always returns the default reason phrase for the status code, whereas HTTP/2 does not support reason phrases at all.
```
Comment (joostverhoog) [inline content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md]: Can we elaborate on this a little bit? Maybe add a section here about "Why does setting the ReasonPhrase have no effect?" and linking to it from the other two places.

That allows us to write an answer that would be understandable to low code developers, too. Something like "In HTTP/2, reason phrases have been removed. Following that, many web servers (including the one that the Mendix runtime uses), do not allow setting anything other than the default reason phrase for the status code, even in HTTP/1, which technically allows it."

Comment (sbont) [reply content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md]: I like that

File: content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md [hunk 2]
```diff
@@ -36,8 +36,12 @@ The `HttpResponse` entity has the following attributes:
...
-
 You can retrieve or create response headers via the `HttpHeaders` association.
 
 An important `HttpResponse` header is `Content-Type`, which indicates how the content should be interpreted. For more information on this header, see the [W3C specification of Content-Type](https://www.w3.org/Protocols/rfc1341/4_Content-Type.html).
+
+### 3.1 Why Does Setting `ReasonPhrase` Have No Effect? {#reason-phrase}
+
+The HTTP/1.x protocol allowed servers to include a __reason phrase__ in the response, including non-standard ones. In HTTP/2, reason phrases have been removed. Following that, many web servers (including the one that the Mendix runtime uses), do not allow setting anything other than the default reason phrase for the status code, even in HTTP/1.x, which technically allows it.
+
+The attribute `ReasonPhrase` is maintained as it is still included when an `HttpResponse` is received from a consumed REST service call.
```
Comment (joostverhoog) [inline content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md]: "As a result, the `ReasonPhrase` attribute is only useful for reading."

(it's not limited to "received from a consumed REST service call")

Comment (sbont) [reply content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md]: Changed.

File: content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md (hunk 1 above)
Comment (joostverhoog) [reply content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md]: Looks good

File: content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md (hunk 2 above)
Comment (joostverhoog) [reply content/en/docs/refguide/modeling/integration/rest-services/http-request-and-response-entities.md]: Looks good

**Pull request merged**